pkg/eosclient/eosbinary/eosbinary.go

```diff
@@ -720,19 +736,18 @@ func (c *Client) Write(ctx context.Context, auth eosclient.Authorization, path s
 	if err != nil {
 		return err
 	}
-
-	return c.WriteFile(ctx, auth, path, fd.Name())
+	return c.WriteFile(ctx, auth, path, fd.Name(), app)
 }
 
 // WriteFile writes an existing file to the mgm.
-func (c *Client) WriteFile(ctx context.Context, auth eosclient.Authorization, path, source string) error {
+func (c *Client) WriteFile(ctx context.Context, auth eosclient.Authorization, path, source, app string) error {
 	xrdPath := fmt.Sprintf("%s//%s", c.opt.URL, path)
 	args := []string{"--nopbar", "--silent", "-f", source, xrdPath}
 
 	if auth.Token != "" {
 		args[4] += "?authz=" + auth.Token
 	} else if auth.Role.UID != "" && auth.Role.GID != "" {
-		args = append(args, fmt.Sprintf("-ODeos.ruid=%s&eos.rgid=%s&eos.app=reva_eosclient::write", auth.Role.UID, auth.Role.GID))
+		args = append(args, fmt.Sprintf("-ODeos.ruid=%s&eos.rgid=%s&eos.app=%s", auth.Role.UID, auth.Role.GID, app))
```

Comment:
The reva_eosclient is the appname used to distinguish traffic from Reva from other traffic (sync, fuse). To include the sub-app that is used to write or read a file we can include it such:

reva_eosclient::write::<app>>

if app is empty, the app should be:
`reva_eosclient::write` and not `reva_eosclient::write::`.

The `write or read` verb is left on purpose as the second argument and on not the third, to avoid having "apps" named "write" or "read" when the app is empty.

Reply:
This is tricky now, the issue being that EOS uses `app` both for traffic tagging and for locking. So on one side we could decide to go for some `reva_eosclient::<app>`  - currently we'd have `reva_<app>`, and the GRPC client does not tag the traffic because this `app` argument was not supported. But surely we can't have `write::<app>` and `read::<app>` in the context of locking.
That said I'd propose a generic `reva::write`, `reva::read` (in case of no locks) and `reva::app_<appname>` in case of locks for the tags to be used everywhere.

Reply:
OK so the last commit does exactly that, by extracting an `EncodeAppName` function at the `eosfs` service level